core/src/test/scala/org/apache/spark/deploy/master/MasterSuite.scala

```diff
@@ -731,6 +731,7 @@ class MasterSuite extends SparkFunSuite
       master.registerApplication(appInfo)
       startExecutorsOnWorkers(master)
       assert(appInfo.executors.map(_._2.worker.id).toSeq.distinct.sorted === expected)
+      System.gc()
```

Comment:
How about adding `System.gc()` in the `after` method.

https://github.com/apache/spark/blob/5a71f218420762bf609087d9c6f7c631fbff1be1/core/src/test/scala/org/apache/spark/deploy/master/MasterSuite.scala#L163-L169